contracts/VoteDelegation.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.12;
+import "./interfaces/IVotingEscrow.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract VoteDelegation is Ownable {
+    struct Delegation {
+        address to;
+        uint256 until;
+    }
+
+    mapping(address => Delegation) public delegation;
+    mapping(address => address[]) public delegated;
+
+    uint256 MAX_DELAGATED = 1_000;
+
+    address public veToken;
+
+    event Delegate(address sender, address recipient, uint256 until);
```

Comment:
Question: do we want to make sender and recipient `indexed`?

Reply:
You should tell me!